pkg/debug/debug.go

```diff
@@ -0,0 +1,144 @@
+package debug
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/spf13/afero"
+)
+
+const (
+	testArtifactsDirectoryVarName = "TEST_ARTIFACTS_DIRECTORY"
+)
+
+var (
+	// globals for overriding in unit tests
+	artifactsDirectoryBase = os.Getenv(testArtifactsDirectoryVarName) // nolint:gochecknoglobals
+	execCommand            = exec.Command                             // nolint:gochecknoglobals
+)
+
+// CollectArtifacts collects useful debugging artifacts from a given namespace.
+// Should typically be called if CurrentGinkgoTestDescription().Failed, like this:
+//   debug.CollectArtifacts(afero.NewOsFs(), GinkgoWriter, TestNamespace, KubeConfigPath, KubectlPath)
+func CollectArtifacts(fs afero.Fs, writer io.Writer, namespace, kubeConfigPath, kubectlPath string) {
+	err := collectNamespacedResources(fs, writer, namespace, kubeConfigPath, kubectlPath)
+	if err != nil {
+		_, _ = fmt.Fprintf(writer, "collection of resources for debugging failed: %v\n", err)
+	}
+}
+
+func collectNamespacedResources(fs afero.Fs, writer io.Writer, namespace, kubeConfigPath, kubectlPath string) error {
+	if artifactsDirectoryBase == "" {
+		return fmt.Errorf("$%s not set", testArtifactsDirectoryVarName)
+	}
+
+	_, _ = fmt.Fprintf(writer, "collecting namespaced resources for debugging...\n")
+
+	cmd := execCommand(
```

Comment:
You should use the `cmd` package of `test-tools` for that. Is there a reason why this needs to call `kubectl` or can this be done using the Kubernetes API instead? I see this as a good opportunity to add the needed types for this to our own `kubernetes` package and dogfood this here.

Reply:
I looked at the `cmd` package and I'm not very fond of it - it does not seem to add any value, makes some `os/exec` features unavailable, introduces its own specific syntax. I'd rather use the standard package.

See point 3 in https://github.com/kudobuilder/test-tools/pull/24#issuecomment-607104384 for why I'm using `kubectl`.

Reply:
What's missing in `cmd` that you need here? Would be great to add it to `cmd` then.